### ISSUE TITLE
Out-of-tree builds put JuliaSyntax symlink in source directory

### ISSUE THREAD

**Issue by maleadt** [JuliaLang/julia#51230]
Despite doing an out-of-tree build (`make O=build/dev configure`), the build system puts a `JuliaSyntax` symlink in the `base` source directory:

```
❯ ls -la base/JuliaSyntax
 base/JuliaSyntax -> build/dev/deps/JuliaSyntax-045d156c44dbb87769c7416d049a7c08908539d4/
```

This can result in surprising behavior when using multiple out-of-tree builds.

Is there a reason the JuliaSyntax stdlib is treated differently from other standard libraries?

**Comment by vtjnash** [comment]
Duplicate of #31140. Fixed by #45441. Reverted by #46420, since people don't like change.

**Comment by KristofferC** [comment]
> Reverted by https://github.com/JuliaLang/julia/pull/46420, since people don't like change.

IIRC the issue was getting duplicated results when grepping in the code base?
